### ISSUE TITLE
rebroadcast want list (should we?)

### ISSUE THREAD

**Issue by pgte** [ipfs/js-ipfs-bitswap#160]
go-ipfs seams to rebroadcast the want list on a fixed interval.
In JS, there is a `rebroadcastDelay` defined in [`src/constants.js`](https://github.com/ipfs/js-ipfs-bitswap/blob/master/src/constants.js), but it's not being used.

Should we implement a rebroadcast?
If all the changes in the waitlist get broadcasted, why would a node get out of sync (unless the transport is somehow unreliable, which would be a more troubling issue...)?

@whyrusleeping do you have any insight on this?

**Comment by dignifiedquire** [comment]
we don‘t use the constant but we do rebroadcast already: https://github.com/ipfs/js-ipfs-bitswap/blob/master/src/want-manager/index.js#L118

**Comment by pgte** [comment]
Why are we rebroadcasting?

**Comment by dignifiedquire** [comment]
> Why are we rebroadcasting?

tldr, because go-ipfs does so. 

I think the idea is to make sure things get eventually resolved if the initial sends don't yield any results. I suspect @whyrusleeping can shed more light on this.

**Comment by dgrisham** [comment]
It's not entirely obvious why we rebroadcast in go-ipfs from the source/comments (or the whitepaper, which mentions it as well). My guess is that it's to resolve any syncing issues that might build up over time with the incremental wantlist updates.

**Comment by whyrusleeping** [comment]
Yeah, the rebroadcasting isn't really needed anymore. It was from a less
reliable time

**Comment by daviddias** [comment]
So the verdict is to remove this rebroadcast, correct?
